lib/shopify-cli/core/monorail.rb

```diff
@@ -104,13 +74,39 @@ def send_event(event)
               use_ssl: ENDPOINT_URI.scheme == 'https'
             ) do |http|
               post = Net::HTTP::Post.new(ENDPOINT_URI.request_uri, headers)
-              post.body = JSON.dump(event)
+              post.body = JSON.dump(build_payload(start_time, end_time, args, err))
               http.request(post)
             end
           rescue
             # silently fail on errors, fire-and-forget approach
```

Comment:
there was some conversation around async retries - where did we land w that?

Reply:
async anything requires a lot more complexity that at this point we are not willing to introduce for this. This all happens at the end of the command so if it does not work and we want to retry we either pause and keep retrying or write it to a file, and then on another command, try to read-parse-post that data as well. At this moment, it's more complexity than we are willing to commit to as this should give us a good sampling.